### PR TITLE
fix: fixes 3 bugs

### PR DESCRIPTION
**Pull request by Bas950**

### CODE REVIEW

File: src/main.mjs [hunk 1]
```diff
@@ -14,6 +15,9 @@ const v6db = setting.v6
 const locFieldHash = setting.locFieldHash
 const mainFieldHash = setting.mainFieldHash
 
+const __filename = fileURLToPath(import.meta.url)
+const __dirname = path.dirname(__filename)
```
Comment (Bas950) [inline src/main.mjs:19]: Solves the issue of getting `__dirname is not defined`

File: src/main.mjs [hunk 2]
```diff
@@ -235,7 +239,7 @@ export const updateDb = (_setting) => {
 	// However, db.js import many external modules, it makes slow down the startup time and uses more memory.
 	// Therefore, we use exec() to run the script in the other process.
 	return new Promise((resolve, reject) => {
-		var cmd = 'node ' + path.resolve(__dirname, '..', 'script', 'updatedb.js')
+		var cmd = 'node ' + path.resolve(__dirname, '..', 'script', 'updatedb.mjs')
```
Comment (Bas950) [inline src/main.mjs:242]: Fixes the wrong file extension

File: src/db.mjs [hunk 3]
```diff
@@ -33,7 +33,12 @@ export const update = async () => {
 	if(refreshTmpDir || !fsSync.existsSync(setting.tmpDataDir)){
 		// refresh tmp folder
 		await rimraf(setting.tmpDataDir)
-		await fs.mkdir(setting.tmpDataDir)
+		await fs.mkdir(setting.tmpDataDir, {recursive: true})
+	}
+
+	// When specifying a custom dataDir, it doesn't always exists
+	if (!fsSync.existsSync(setting.dataDir)){
+		await fs.mkdir(setting.dataDir, {recursive: true})
```
Comment (Bas950) [inline src/db.mjs:41]: Creates the directory when needed

**Comment by sapics** [comment]
Thank you for fixing bugs!

**Pull request merged**